ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py

```diff
@@ -336,6 +336,34 @@ def test_dropout1d(
         assert u.shape == v.shape == w.shape
 
 
+@handle_test(
+    fn_tree="functional.ivy.experimental.separable_conv2d",
+    ground_truth_backend="jax",
+    x_k_s_p=helpers.arrays_for_pooling(min_dims=2, max_dims=2, min_side=1, max_side=4),
```

Comment:
Inside `arrays_for_pooling`, you are not generating the appropriate `array_dims` and so the `kernel` is not initialized at all. You are passing `max_dims = 2`. I suggest you look at the helper function inside the file, you'll get an idea.